owncloudApp/src/main/java/com/owncloud/android/providers/FileContentProvider.kt

```diff
@@ -0,0 +1,1470 @@
+                                db,
+                                ContentUris.withAppendedId(ProviderTableMeta.CONTENT_URI_FILE, childId), null, null
+                            )
+                        }
+                        children.moveToNext()
+                    }
+                    children.close()
+                }
+                count += db.delete(
+                    ProviderTableMeta.FILE_TABLE_NAME,
+                    ProviderTableMeta._ID
+                            + "="
+                            + uri.pathSegments[1]
+                            + if (!TextUtils.isEmpty(where))
+                        " AND (" + where
+                                + ")"
+                    else
+                        "", whereArgs
+                )
+            }
+            ROOT_DIRECTORY ->
+                //Log_OC.d(TAG, "Removing ROOT!");
+                count = db.delete(ProviderTableMeta.FILE_TABLE_NAME, where, whereArgs)
+            SHARES -> count = db.delete(ProviderTableMeta.OCSHARES_TABLE_NAME, where, whereArgs)
+            CAPABILITIES -> count = db.delete(ProviderTableMeta.CAPABILITIES_TABLE_NAME, where, whereArgs)
+            UPLOADS -> count = db.delete(ProviderTableMeta.UPLOADS_TABLE_NAME, where, whereArgs)
+            CAMERA_UPLOADS_SYNC -> count = db.delete(ProviderTableMeta.CAMERA_UPLOADS_SYNC_TABLE_NAME, where, whereArgs)
+            QUOTAS -> count = db.delete(ProviderTableMeta.USER_QUOTAS_TABLE_NAME, where, whereArgs)
+            else -> throw IllegalArgumentException("Unknown uri: " + uri.toString())
+        }
+        return count
+    }
+
+    override fun getType(uri: Uri): String? {
+        when (mUriMatcher!!.match(uri)) {
+            ROOT_DIRECTORY -> return ProviderTableMeta.CONTENT_TYPE
+            SINGLE_FILE -> return ProviderTableMeta.CONTENT_TYPE_ITEM
+            else -> throw IllegalArgumentException("Unknown Uri id." + uri.toString())
+        }
+    }
+
+    override fun insert(uri: Uri, values: ContentValues): Uri? {
+        var newUri: Uri?
+        val db = mDbHelper!!.writableDatabase
+        db.beginTransaction()
+        try {
+            newUri = insert(db, uri, values)
+            db.setTransactionSuccessful()
+        } finally {
+            db.endTransaction()
+        }
+        context!!.contentResolver.notifyChange(newUri!!, null)
+        return newUri
+    }
+
+    private fun insert(db: SQLiteDatabase, uri: Uri, values: ContentValues): Uri {
+        when (mUriMatcher!!.match(uri)) {
+            ROOT_DIRECTORY, SINGLE_FILE -> {
+                val remotePath = values!!.getAsString(ProviderTableMeta.FILE_PATH)
+                val accountName = values.getAsString(ProviderTableMeta.FILE_ACCOUNT_OWNER)
+                val projection =
+                    arrayOf(ProviderTableMeta._ID, ProviderTableMeta.FILE_PATH, ProviderTableMeta.FILE_ACCOUNT_OWNER)
+                val where = ProviderTableMeta.FILE_PATH + "=? AND " +
```

Comment:
`val where = "${ProviderTableMeta.FILE_PATH}=? AND ${ProviderTableMeta.FILE_ACCOUNT_OWNER}=?"
`
is that definition clearer to you?

Reply:
Yessss